### PR TITLE
Use official jamm

### PR DESCRIPTION
**Pull request by Abacn**
According to the comment mark
```
// Swap to use the officially published version of 0.4.x once available
// instead of relying on a community published copy. See
// https://github.com/jbellis/jamm/issues/44 for additional details.
```
Offical jamm 0.4.0 has been released in July 2023: https://mvnrepository.com/artifact/com.github.jbellis/jamm/0.4.0

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/harness/src/main/java/org/apache/beam/fn/harness/Caches.java [hunk 1]
```diff
@@ -65,8 +65,7 @@ public final class Caches {
    */
   private static final long CACHE_SIZE_CHANGE_LIMIT_BYTES = 1 << 16;
 
-  private static final MemoryMeter MEMORY_METER =
-      MemoryMeter.builder().withGuessing(Guess.BEST).build();
```
Comment (Abacn) [inline sdks/java/harness/src/main/java/org/apache/beam/fn/harness/Caches.java:69]: The API of official jamm 0.4 has adjusted to be different than the community one. BEST is the default for Builder and its not an enum: https://github.com/jbellis/jamm/blob/4b475b906d9a5dcef523db49124fbb8f68e67ee0/src/org/github/jamm/MemoryMeter.java#L589

**Comment by Abacn** [comment]
Java PreCommit stuck test `org.apache.beam.runners.fnexecution.logging.GrpcLoggingServiceTest. testMultipleClientsFailingIsHandledGracefullyByServer` not related

**Comment by Abacn** [comment]
assign set of reviewers

**Comment by damccorm** [comment]
> Java PreCommit stuck test `org.apache.beam.runners.fnexecution.logging.GrpcLoggingServiceTest. testMultipleClientsFailingIsHandledGracefullyByServer` not related

Doesn't look like this is normally flaky - https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java.yml?query=event%3Aschedule - so lets get a green signal before merging (kicking off a retry)

**Comment by Abacn** [comment]
> GrpcLoggingServiceTest

The rerun is green

I saw the timeout test observed before: #14474 it has higher probability when gradle cache is disabled. Similarly cron jobs appear to be less flaky than the PR that changed java core code.

**Review by damccorm — APPROVED** [review]
LGTM - thanks

**Comment by Abacn** [comment]
Also manually tested a dataflow wordcount `2024-01-19_11_04_21-10729576904645109680`, checked the worker log there is no warning about jamm, and the job succeeded

merging for now

(there is a followup that build container locally see error that `command --provenance does not exist`, Add a fix in #30051)

**Pull request merged**